src/components/settings-modal/settings-modal.js

```diff
@@ -65,6 +74,7 @@ const SettingsModal = ({
 
         onToggleIsPrettyName(isPrettyNameValue);
         onToggleShowFeatureHints(showFeatureHintsValue);
+        onToggleShowDatasetPreviews(showDatasetPreviewsValue);
```

Comment:
do you think we can have the  update state `onToggleShowDatasetPreviews(showDatasetPreviewsValue);` and API call together? 

perhaps something here like 

`updateShowDatasetPreviews()`

```
const updateShowDatasetPreviews = () => {
    // to update redux state
    onToggleShowFeatureHints(showFeatureHintsValue);

   // to update BE
   const updatePreferences = async () => {
      try {
        const response = await fetch('/api/preferences', {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            showDatasetPreviews: showDatasetPreviewsValue,
          }),
        });

        if (!response.ok) {
          throw new Error('Failed to update preferences');
        }

        const result = await response.json();
        console.log('Preferences updated successfully:', result);
      } catch (error) {
        console.error('Error updating preferences:', error);
      }
    };

    updatePreferences();
}
```

It's just easier to know that all the actions related to update dataset preview are in one place

Reply:
The API call moved to a separate file in `src/utils/preferences-api.js`.